### PR TITLE
AlienFlight F7 support

### PR DESCRIPTION
**Pull request by MJ666**
Don't merge, still in testing. Requires the AlienFlight F4 support #1589 merged first and likely @sambas F722 support #1680 will also be useful to be merged in advance.

### CODE REVIEW

**Comment by MJ666** [comment]
Rebased, but not flying yet. Everything looks good at the bench, it takes of normaly but its trying to flip with any roll or pitch input. Need some more testing.

**Comment by sambas** [comment]
stm32f7xx_hal_conf.h in target folder not needed anymore

**Comment by MJ666** [comment]
@sambas thanks for the heads up. Removed the file.

**Comment by MJ666** [comment]
Updated and rebased.

File: src/main/target/ALIENFLIGHTNGF7/target.h [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+#define SPI3_MISO_PIN           PB4
+#define SPI3_MOSI_PIN           PB5
+
+#define USE_I2C
+#define USE_I2C_PULLUP
+#define I2C_DEVICE              (I2CDEV_1)
+#define I2C1_SCL                PB6
+#define I2C1_SDA                PB7
+
+#define USE_ADC
```
Comment (digitalentity) [inline src/main/target/ALIENFLIGHTNGF7/target.h:156]: @MJ666 please update ADC usage to latest changes:
```
#define USE_ADC
#define ADC_CHANNEL_1_PIN               PC0
#define ADC_CHANNEL_2_PIN               PC1
#define ADC_CHANNEL_3_PIN               PC4
#define VBAT_ADC_CHANNEL                ADC_CHN_1
#define CURRENT_METER_ADC_CHANNEL       ADC_CHN_2
#define RSSI_ADC_CHANNEL                ADC_CHN_3
```

**Review by digitalentity — CHANGES_REQUESTED** [review]

File: src/main/target/ALIENFLIGHTNGF7/target.h (hunk 1 above)
Comment (MJ666) [reply src/main/target/ALIENFLIGHTNGF7/target.h:156]: Thanks for the heads up. Updated.

**Review by digitalentity — APPROVED** [review]

**Comment by digitalentity** [comment]
@MJ666 Thanks for quick changes. I'm merging this now

**Pull request merged**